resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolvers.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver.dns;
+
+import io.netty.channel.ChannelFactory;
+import io.netty.channel.EventLoop;
+import io.netty.channel.ReflectiveChannelFactory;
+import io.netty.channel.socket.DatagramChannel;
+import io.netty.channel.socket.InternetProtocolFamily;
+import io.netty.resolver.HostsFileEntriesResolver;
+
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+public final class DnsNameResolvers {
+
+    /**
+     * Creates a new builder.
+     *
+     * @param eventLoop the {@link EventLoop} the {@link EventLoop} which will perform the communication with the DNS
+     * servers.
+     * @return a new {@link Builder} instance
+     */
+    public static Builder newDnsNameResolver(EventLoop eventLoop) {
+        return new Builder(eventLoop);
+    }
+
+    /**
+     * Can't be instanciated.
+     */
+    private DnsNameResolvers() {
+    }
+
+    /**
+     * A {@link DnsNameResolver} builder.
+     */
+    public static final class Builder {
```

Comment:
OK. I usually prefer having a static bootstrap method, but if that's not the current coding style in Netty, let's keep things consistent.